### PR TITLE
Extract the hostname if input is a url, and do this before the ip check

### PR DESCRIPTION
**Pull request by erikvold**

### CODE REVIEW

**Comment by wbnns** [comment]
@erikvold Hi Erik! Apologies for the delay on this. Thanks a bunch for submitting this PR. We'll review it ASAP and will keep you posted here.

File: ping21.py [hunk 1]
```diff
@@ -7,9 +7,25 @@
...
+
+    Returns:
+        str: cleaned uri
+
+    """
+    try:
+        hostname = urlparse(uri).hostname
+        if hostname is not None:
+            return hostname
+    except ValueError:
```
Comment (j2kun) [inline ping21.py]: urlparse doesn't raise a ValueError, so no try/except needed here.

**Comment by j2kun** [comment]
@erikvold Thanks for this! I left a minor inline comment, and I'll be happy to merge it once that's fixed.

One optional note: the one case that one might want to support (but currently isn't supported) is 

`21 buy "http://IP:PORT/?uri=21.co/learn"`

which produces

```
ping: cannot resolve 21.co/learn: Unknown host
HTTP Status 400: ping cannot be performed on url=21.co/learn
```

If you're interested we could also add that case to this PR, which we could do by adding a scheme if none is present. This might be as simple as

```
def getHostname(uri):
   if 'http' != uri[:4]:
      uri = 'http://' + uri
   return urlparse(uri).hostname
```

**Comment by erikvold** [comment]
Thanks @j2kun and @wbinns!

I've made the updates that were requested.

**Comment by j2kun** [comment]
LGTM

**Comment by wbnns** [comment]
Thanks again, @erikvold!

**Pull request merged**